controlplane/kubeadm/internal/controllers/remediation.go

```diff
@@ -49,22 +50,35 @@ func (r *KubeadmControlPlaneReconciler) reconcileUnhealthyMachines(ctx context.C
 	errList := []error{}
 	healthyMachines := controlPlane.HealthyMachinesByMachineHealthCheck()
 	for _, m := range healthyMachines {
-		if conditions.IsTrue(m, clusterv1.MachineHealthCheckSucceededCondition) &&
-			conditions.IsFalse(m, clusterv1.MachineOwnerRemediatedCondition) &&
-			m.DeletionTimestamp.IsZero() {
-			patchHelper, err := patch.NewHelper(m, r.Client)
-			if err != nil {
-				errList = append(errList, err)
-				continue
-			}
+		if !m.DeletionTimestamp.IsZero() {
+			continue
+		}
+
+		shouldCleanup := conditions.IsTrue(m, clusterv1.MachineHealthCheckSucceededCondition) && conditions.IsFalse(m, clusterv1.MachineOwnerRemediatedCondition)
+		shouldCleanupV1Beta2 := v1beta2conditions.IsTrue(m, clusterv1.MachineHealthCheckSucceededV1Beta2Condition) && v1beta2conditions.IsFalse(m, clusterv1.MachineOwnerRemediatedV1Beta2Condition)
+
+		if !(shouldCleanup || shouldCleanupV1Beta2) {
+			continue
+		}
+
+		patchHelper, err := patch.NewHelper(m, r.Client)
+		if err != nil {
+			errList = append(errList, err)
+			continue
+		}
 
+		if shouldCleanup {
 			conditions.Delete(m, clusterv1.MachineOwnerRemediatedCondition)
+		}
 
-			if err := patchHelper.Patch(ctx, m, patch.WithOwnedConditions{Conditions: []clusterv1.ConditionType{
-				clusterv1.MachineOwnerRemediatedCondition,
-			}}); err != nil {
-				errList = append(errList, err)
-			}
+		if shouldCleanupV1Beta2 {
+			v1beta2conditions.Delete(m, clusterv1.MachineOwnerRemediatedV1Beta2Condition)
+		}
+
+		if err := patchHelper.Patch(ctx, m, patch.WithOwnedConditions{Conditions: []clusterv1.ConditionType{
+			clusterv1.MachineOwnerRemediatedCondition,
```

Comment:
> We have to set these options based on shouldCleanup & shouldCleanupV1Beta2 (+ of course also include the v1beta2 condition)

This is not necessary,  if we don't change the field the corresponding option is no op

> Can you please audit for all places where we use WithOwnedConditions today if we forgot to add the v1beta2 version? (for the PRs where we already implemented v1beta2)

We should be ok
I have on my todo list to get back and check the MHC controller, but there should not be impact, considering that the option only helps to address config when two controllers try to set the same condition. More specifically
- MHC is the only controller setting the HealthCheckSucceded condition
- MHC only creates OwnerRemediated condition if it does not exists, then ignores it

Probably we can improve assigning clear ownership of the conditions above, but that I don't expect any impact on the behaviour (because there are no conflicts)